scripts/init.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+// Sets the start script for react-wp-scripts
+// And move the loader.php to the project root folder
+process.on( 'unhandledRejection', err => {
+	throw err;
+} );
+
+const fs = require( 'fs-extra' );
+const path = require( 'path' );
+const chalk = require( 'chalk' );
+const upperCamelCase = require( 'uppercamelcase' );
+
+module.exports = function(
+	appPath,
+	appName,
+	verbose,
+	originalDirectory,
+	template
+) {
+
+	// Parse a namespace based on the name of the package
+	let namespace = 'ReactWPScripts';
+	try {
+		const projectPackageJSON = require( path.join( process.cwd(), 'package.json' ) );
+		namespace = upperCamelCase( projectPackageJSON.name );
+	}
+	catch ( err ) {}
+
+	const pkgName = require( path.join( __dirname, '..', 'package.json' ) ).name;
+	const reactWPScriptsPath = path.join( appPath, 'node_modules', pkgName );
+	const appPackage = require( path.join( appPath, 'package.json' ) );
+
+	const scriptsPath = path.resolve(
+		process.cwd(),
+		'node_modules',
+		'react-scripts',
+		'scripts',
+		'init.js'
+	);
+	const reactScriptsInit = require(scriptsPath);
+	reactScriptsInit( appPath, appName, verbose, originalDirectory, template );
+
+	// Setup the custom start script
+	appPackage.scripts.start = 'react-wp-scripts start';
+
+	fs.writeFileSync(
+		path.join( appPath, 'package.json' ),
+		JSON.stringify( appPackage, null, 2 )
+	);
+
+	// Copy the loader.php
+	const loaderPath = path.join( reactWPScriptsPath, 'loader.php' );
+
+	function successMessage() {
+		console.log( chalk.green( 'React WP Scripts Loader copied to your project root folder.' ) );
+		console.log( chalk.green( 'Please, follow instructions to add PHP to enqueue your assets:' ) );
+		console.log( chalk.blue( 'https://github.com/humanmade/react-wp-scripts#react-wp-scripts' ) );
+	}
+
+	const destinationFile = path.join( appPath, 'react-wp-scripts.php' );
+	fs.copy( loaderPath, destinationFile )
+		.then( () => {
+			// Replace %%NAMESPACE%% for the specified namespace
+			fs.readFile( destinationFile, 'utf8', function( err, data ) {
+				if ( err ) {
+					console.log( err );
+				}
```

Comment:
I think the error handling here is a bit inconsistent. If we consider a read error here to be a fatal error, we should `throw` that error so that it will be picked up by the catch below, or at minimum return after we log it to the console.

We also don't return a promise from this first `.then`, so the success message will likely display before this action has necessarily completed.

This is untested but I think we'd be looking for something more like this:
```
fs.copy( loaderPath, destinationFile )
  .then( () => new Promise(( resolve, reject ) => {
    // Replace %%NAMESPACE%% for the specified namespace
    fs.readFile( destinationFile, 'utf8', function( err, data ) {
      if ( err ) {
        return reject( err );
      }

      var result = data.replace( '%%NAMESPACE%%', namespace );
      fs.writeFile( destinationFile, result, 'utf8', function( err ) {
        if ( err ) {
          return reject( err );
        }
        resolve();
      } );
    } );
  } ) )
  .then( () => successMessage() )
  .catch( err => {
    console.log( chalk.bgRed( 'React WP Scripts loader could not be copied to your root folder. Error details:' ) );
    console.log( chalk.red( err ) );
  } );
```